index.qmd

```diff
@@ -1,120 +1,410 @@
+
+Dataset can be found here: (<https://drive.google.com/file/d/1eAplOYO-k7ZYHj4uHAY1tEr8VTeaxS6u/view?usp=sharing>).
+
+### Required Steps to Build Model
+
+1.  Load necessary packages
+
+2.  Load and explore the dataset
+
+3.  Data visualization and exploratory analysis
 
-Refer to output in your narrative like this: x=`r x` .
+4.  Preprocess data and Train model
 
-Load any required packages in a code chunk (you may need to install some packages):
+5.  Evaluate model
 
-```{r, message=F, warning=F}
+6.  Make prediction with case study
+
+7.  ROC Curve
+
+### Load Required Packages
+
+Load necessary r packages to aid analysis.
+
+```{r, message=F, warning=FALSE, results=FALSE}
+#install.packages("corrplot")
+#install.packages("caret")
+#install.packages("kableExtra")
+ #install.packages("tinytex")
+#install.packages("DT")
+#install.packages("heatmaply")
+library(tinytex)
 library(tidyverse)
 library(leaflet)
 library(kableExtra)
 library(htmlwidgets)
 library(widgetframe)
+library(dplyr)
+library(tidyr)
+library(forcats)
+library(ggplot2)
+library(class)
+library(corrplot)
+library(caret)
+library(reshape2) #for melt function
+library(rmarkdown) 
+library(knitr)
+library(pROC) #for ROC Curve
+library(widgetframe)
+library(DT)
+library(heatmaply)
+library(plotly)
 knitr::opts_chunk$set(widgetframe_widgets_dir = 'widgets' ) 
 knitr::opts_chunk$set(cache=TRUE)  # cache the results for quick compiling
 ```
 
-## Download and clean all required data
+```{r include=FALSE}
+# Load renv
+library(renv)
+
+# Check status
+renv::status()
+
+# Synchronize the project
+renv::snapshot()
+```
+
+### Load and Explore Data
 
 ```{r}
-n=20
-data=data.frame(x=runif(n,-180,180),
-                y=runif(n,-60,60),
-                size = runif(n, 5, 20),
-                category = factor(
-                  sample(letters[1:5], n, replace = TRUE)
-                  ),
-                value = rnorm(n))
+diabetes_data <- read.csv("https://drive.google.com/uc?export=download&id=1eAplOYO-k7ZYHj4uHAY1tEr8VTeaxS6u")
+
 ```
 
+Exploring the dataset
+
 ```{r, results='asis'}
-data %>% 
+diabetes_data%>%
   slice(1:10) %>% #show only 1:n rows
   kable(digits=2,align="c")%>% #make table and round to two digits
   kable_styling(bootstrap_options = 
-                  c("striped", "hover", "condensed", "responsive")) #apply other formatting
+                  c("striped", "hover", "condensed", "responsive")) 
+
+
+```
+
+```{r}
+#Exploring the structure of the data
+str(diabetes_data)
+  
+```
+
+The dataset contains 15000 patient entries, with all features being numeric values.
+
+##### Clean Dataset
+
+```{r results='hide'}
+duplicated(diabetes_data) #check for duplicates
+sum(is.na(diabetes_data)) #to check missing values 
+
 ```
 
-Add any additional processing steps here.
+The dataset contains no duplicates or missing values. The next step is visual summary.
+
+### Data Visualization and Exploratory Analysis
+
+Correlation and visual summary
```

Comment:
correlation and multicollinearity aren't quite the same. Since you're checking VIF between variables, I believe multicollinearity might be more suitable